client/src/components/recipes-collections/SingleRecipeView.jsx

```diff
@@ -0,0 +1,118 @@
+import {useEffect, useState} from "react";
+import {useParams} from "react-router-dom";
+import {Interweave} from "interweave";
+import {parseIngredient} from "parse-ingredient";
+import spinner from "./spinner.svg";
+import {ReactSVG} from "react-svg";
+import RecipeNutritionLabel from "./RecipeNutritionLabel";
+
+const SingleRecipeView = ({siteUser}) => {
+  const {recipeId} = useParams();
+  const [selectedRecipe, setSelectedRecipe] = useState({});
+  const [ingredients, setIngredients] = useState([]);
+
+  useEffect(() => {
+    // Fetch data for a single recipe
+    const viewRecipe = async (recipeId) => {
+      await fetch(`/api/recipe/${recipeId}`)
+        .then((res) => res.json())
+        .then((res) => {
+          const parsedIngredients = [...parseIngredient(res[0].ingredients)];
+          setSelectedRecipe({...res[0], ingredients: parsedIngredients});
+          console.log("parsed ingredients: ", parsedIngredients);
+          setIngredients(
+            parsedIngredients
```

Comment:
Nit: Instead of doing this work inside of the `setState`, you could pull out this logic into its own function, and then call that function on your `parsedIngredients` data. This has a couple of benefits:
- It makes the code a bit more readable, since you have to keep less context in your head
- When do you end up wanting to test your code, you can test the isolated logic in that function instead of the whole file
- It becomes easier to reuse

Reply:
Done! Thank you!